src/main/java/com/scalar/cassy/transferer/AzureBlobFileUploader.java

```diff
@@ -66,10 +67,17 @@ public void upload(List<Path> files, BackupConfig config) {
       } else {
         logger.info(filePath + " has been already uploaded.");
       }
+      if (toBeUploaded.size() % asyncFileUploadLimit == 0) {
+        // Start asynchronously all uploads and wait for all of them to complete
```

Comment:
```suggestion
        // Start uploading files asynchronously and wait for them to complete
```

Reply:
Thank you for your suggestions.
Suggestions updated accordingly.